The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/11259651690/job/31309145243) [(plain)](https://github.com/rust-lang-ci/rust/commit/aace1d596a23bb68394dc48448bef4526b407922/checks/31309145243/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
.......... (130/133)
.FF        (133/133)


/checkout/tests/rustdoc-gui/methods-left-margin.goml methods-left-margin... FAILED
[ERROR] `tests/rustdoc-gui/methods-left-margin.goml` line 17: The following errors happened: [different X values: 25 (or 25) != 15]: for command `assert-position: (".impl-items section.method", {"x": |x|}, ALL)`
/checkout/tests/rustdoc-gui/notable-trait.goml notable-trait... FAILED
/checkout/tests/rustdoc-gui/notable-trait.goml notable-trait... FAILED
[ERROR] `tests/rustdoc-gui/notable-trait.goml` line 23
    from `tests/rustdoc-gui/notable-trait.goml` line 38
    from `tests/rustdoc-gui/notable-trait.goml` line 87: The following errors happened: [different X values: 15 (or 15) != 25]: for command `assert-position: (
            "//*[@id='method.create_an_iterator_from_read']//a[normalize-space()='NotableStructWithLongName']",
            {"x": |x|},
        )`
[ERROR] `tests/rustdoc-gui/notable-trait.goml` line 27
    from `tests/rustdoc-gui/notable-trait.goml` line 38
    from `tests/rustdoc-gui/notable-trait.goml` line 87: The following errors happened: [different X values: 293.421875 (or 293) != 303]: for command `assert-position: (
            "//*[@id='method.create_an_iterator_from_read']//*[@class='tooltip']",
            {"x": |i_x|},

Error: ()
Build completed unsuccessfully in 0:03:52
  local time: Wed Oct  9 17:34:46 UTC 2024
```

</details>

--- a/src/librustdoc/html/static/css/rustdoc.css
+++ b/src/librustdoc/html/static/css/rustdoc.css
@@ -2435,7 +2435,7 @@ in src-script.js and main.js
 	}
 
 	/* Position of the "[-]" element. */
-	details.toggle:not(.top-doc) > summary {
+	details.toggle:not(.top-doc) > summary, .impl-items > section  {
 		margin-left: 10px;
 	}
 	.impl-items > details.toggle > summary:not(.hideme)::before,

--- a/tests/rustdoc-gui/methods-left-margin.goml
+++ b/tests/rustdoc-gui/methods-left-margin.goml
@@ -1,0 +1,17 @@
+// This test is to ensure that methods are correctly aligned on the left side.
+
+go-to: "file://" + |DOC_PATH| + "/test_docs/struct.Foo.html"
+
+// First we ensure that we have methods with and without documentation.
+assert: ".impl-items > details.method-toggle > summary > section.method"
+assert: ".impl-items > section.method"
+
+// Checking on desktop.
+set-window-size: (900, 600)
+store-position: (".impl-items section.method", {"x": x})
+assert-position: (".impl-items section.method", {"x": |x|}, ALL)
+
+// Checking on mobile.
+set-window-size: (600, 600)
+store-position: (".impl-items section.method", {"x": x})
+assert-position: (".impl-items section.method", {"x": |x|}, ALL)

--- a/tests/rustdoc-gui/methods-left-margin.goml
+++ b/tests/rustdoc-gui/methods-left-margin.goml
@@ -8,10 +8,12 @@ assert: ".impl-items > section.method"
 
 // Checking on desktop.
 set-window-size: (900, 600)
+wait-for-size: ("body", {"width": 900})
 store-position: (".impl-items section.method", {"x": x})
 assert-position: (".impl-items section.method", {"x": |x|}, ALL)
 
 // Checking on mobile.
 set-window-size: (600, 600)
+wait-for-size: ("body", {"width": 600})
 store-position: (".impl-items section.method", {"x": x})
 assert-position: (".impl-items section.method", {"x": |x|}, ALL)

--- a/tests/rustdoc-gui/notable-trait.goml
+++ b/tests/rustdoc-gui/notable-trait.goml
@@ -85,8 +85,8 @@ call-function: ("check-notable-tooltip-position", {
 // Checking on mobile now.
 set-window-size: (650, 600)
 call-function: ("check-notable-tooltip-position-complete", {
-    "x": 15,
-    "i_x": 293,
+    "x": 25,
+    "i_x": 303,
     "popover_x": 0,
 })
 

--- a/tests/rustdoc-gui/notable-trait.goml
+++ b/tests/rustdoc-gui/notable-trait.goml
@@ -84,6 +84,7 @@ call-function: ("check-notable-tooltip-position", {
 
 // Checking on mobile now.
 set-window-size: (650, 600)
+wait-for-size: ("body", {"width": 650})
 call-function: ("check-notable-tooltip-position-complete", {
     "x": 25,
     "i_x": 303,

